### PR TITLE
[BEAM-6428] Add a textual selection syntax for schema fields.

### PR DESCRIPTION
**Pull request by reuvenlax**
This allows the user to write code like Select.fieldNames("a.b", "c.d.*"))

### CODE REVIEW

**Comment by apilloud** [comment]
Run Java PreCommit

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldAccessDescriptor.java [hunk 1]
```diff
@@ -245,43 +452,39 @@ private static Schema getFieldSchema(Field field) {
       return type.getMapValueType().getRowSchema();
     } else {
       throw new IllegalArgumentException(
-          "Field " + field + " must be either a row or " + " a container containing rows");
+          "FieldType " + type + " must be either a row or " + " a container containing rows");
```
[the remark this answers is not in the document]
Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/FieldAccessDescriptor.java]: done

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java [hunk 2]
```diff
@@ -97,89 +103,191 @@ private Select(FieldAccessDescriptor fieldAccessDescriptor) {
...
+              input.getValue(fieldName),
+              nested.getValue(),
+              nestedInputType,
+              nestedOutputType);
+      output.addValue(value);
+    }
+    return output.build();
+  }
+
+  @SuppressWarnings("unchecked")
```
[the remark this answers is not in the document]
Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java]: It's due to the unchecked casts to List and Map below

File: sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java [hunk 3]
```diff
@@ -97,89 +103,191 @@ private Select(FieldAccessDescriptor fieldAccessDescriptor) {
...
+                    entry.getValue(),
+                    fieldAccessDescriptor,
+                    nestedInputType,
+                    nestedOutputType);
+            selectedMap.put(entry.getKey(), selected);
+          }
+          return selectedMap;
+        }
+      default:
+        throw new RuntimeException("Unexpected.");
```
[the remark this answers is not in the document]
Comment (reuvenlax) [reply sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/transforms/Select.java]: done

**Comment by tweise** [comment]
@reuvenlax this PR should have been squash merged or cleaned up otherwise before merging. See https://beam.apache.org/contribute/committer-guide/#finishing-touches

**Pull request merged**